source/agora/test/Base.d

```diff
@@ -812,6 +868,7 @@ public class TestAPIManager
     {
         try
         {
+            this.enrollNonGenesisValidators();
```

Comment:
I dislike shoehorning this stuff into this method. It's explicitly used for discovery of the genesis validators, it shouldn't create any transactions or blocks.

Reply:
Do you suggest we call it from the test directly after network discovery? I wanted to prevent extra calls from all the tests for the completion of test setup.